### PR TITLE
Add image scraping support

### LINKED ISSUE #344: [Feature] Add image support for scrapers
**Describe the solution you'd like**
The performer scrapers should support scraping an image to populate the performer image. 

The scene scraper should support scraping and setting a cover image.

### PR DESCRIPTION
**Pull request by WithoutPants**
Resolves #344 

Adds the ability to scrape performer images and scene cover images.

This change also introduces the `subScraper` xpath post-processing option. If `subScraper` appears in an attribute xpath configuration, then the sub-scraper will be executed after all other post-processes are complete. It then takes the value and performs an http request, using the value as the URL. Within the `subScraper` config is a nested scraping configuration. This allows you to traverse to other webpages to get the attribute value you are after.

For example, from the Boobpedia scraper config in #333 :

```
...
performerScraper:
  performer:
    # ..snip..
    Image:
      selector: //table[@class="infobox"]//tr[2]//a/@href
      # URL is a partial url, add the first part
      replace:
        - regex: ^
          with: http://www.boobpedia.com
      subScraper:
        selector: //div[@class="fullImageLink"]/a/@href
        replace:
          - regex: ^
            with: http://www.boobpedia.com
```

This fragment gets the URL from the xpath `//table[@class="infobox"]//tr[2]//a/@href`, adds the `http://www.boobpedia.com` prefix with the `replace` post-process. Then the sub-scraper post-process is run. It requests the document from the resulting URL, then gets the URL from `//div[@class="fullImageLink"]/a/@href` of the resulting page, followed by the replace post-process.

The `Image` value is expected to be a URL itself, which the system will subsequent request and encode.

Also adds image scraping to the stash scraper.

### CODE REVIEW

File: pkg/scraper/image.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	s.Image = img
+
+	return nil
+}
+
+func getImage(url string) (*string, error) {
+	// assume is a URL for now
+	resp, err := http.Get(url)
```
Comment (bnkai) [inline pkg/scraper/image.go]: should we use a new http client with a default appropriate timeout instead of a plain get ?

**Comment by bnkai** [comment]
Tests ok with me.
Images are fetched ok and saved with the boobpedia demo scraper.

**Comment by WithoutPants** [comment]
Rebased and ported UI changes to 2.5. @bnkai can you please review on v2 and v2.5 of the UI?

**Comment by MrX292** [comment]
https://github.com/stashapp/CommunityScrapers/pull/2 some scrapers for it

**Pull request merged**